### PR TITLE
Normalize TBN

### PR DESCRIPTION
**Pull request by querielo**
Tangen, bitangent and normal have to be orthonormal.

This MR does not fix https://github.com/playcanvas/engine/issues/2787 but improves it.
It looks like Playcanvas Editor has to keep tangents/bitangents while converting fbx=>glb

- Before: 
Screen Shot 2022-07-01 at 17 43 09

- After:
Screen Shot 2022-07-01 at 17 43 14

Tangent visualization of a test model:

- Before:
Screen Shot 2022-07-01 at 16 39 07

- After:
Screen Shot 2022-07-01 at 16 39 11

It looks like there is still something wrong with TBN matrix.

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/graphics/program-lib/chunks/lit/frag/TBNderivative.js [hunk 1]
```diff
@@ -17,9 +17,10 @@ void getTBN() {
...
     vec3 B = dp2perp * duv1.y + dp1perp * duv2.y;
 
-    // construct a scale-invariant frame
-    float denom = max( dot(T,T), dot(B,B) );
-    float invmax = (denom == 0.0) ? 0.0 : tbnBasis / sqrt( denom );
-    dTBN = mat3(T * invmax, -B * invmax, dVertexNormalW );
+    dTBN = mat3(
+        normalize(T * tbnBasis),
+        normalize(-B * tbnBasis),
+        normalize(dVertexNormalW)
```
Comment (slimbuck) [inline src/graphics/program-lib/chunks/lit/frag/TBNderivative.js:23]: no need here - dVertexNormalW is already normalised

Comment (querielo) [reply src/graphics/program-lib/chunks/lit/frag/TBNderivative.js:23]: Oh, I see `dVertexNormalW = normalize(vNormalW);`

**Comment by willeastcott** [comment]
Great to see you trying to improve the quality here!

This PR is deviating the code away from the implementation linked [here](http://www.thetenthplanet.de/archives/1180) at the top of the chunk. It feels like there should be code comments justifying the divergence.

**Comment by querielo** [comment]
> This PR is deviating the code away from the implementation linked [here](http://www.thetenthplanet.de/archives/1180) at the top of the chunk. It feels like there should be code comments justifying the divergence.

@willeastcott It looks like you are right. So, I close this MR.

For me, it looks like TBN are changed manually with rotating tangents (I don't know how exactly it is done). And normal maps are generated on changed tangents. So, Engine cannot reconstruct them after Editor removes tangents while importing fbx to glb.

**Pull request closed without merging**